Docs 1.0 refactor rs (#89)
neuralmagic/docs: src/content/products/sparseml.mdx

## Overview

SparseML is a toolkit that includes APIs, CLIs, scripts and libraries that enable you to create sparse models trained on your data.

> SparseML provides two avenues to accomplish this goal: 

Reviewer: ```suggestion
SparseML provides two options to accomplish this goal: 
```

- **Sparse Transfer Learning**: Fine-tune state-of-the-art pre-sparsified models from the SparseZoo onto your dataset while preserving sparsity.

- **Sparsifying from Scratch**: Apply state-of-the-art [sparsification](https://docs.neuralmagic.com/main/source/getstarted.html#sparsification) algorithms such as pruning and quantization to any neural network.

These options are useful for different situations:

> - **Sparse Transfer Learning** is the easiest path to creating a sparse model trained on your data. Pull down a sparse model from SparseZoo and point our training scripts at your data without any hyperparameter search. This is the recommended pathway for supported use cases like Image Classification, Object Detection, and several NLP tasks. 

Reviewer: @rsnm2 In certain places, these use cases are cited as proper nouns e.g., "Image Classification" but then elsewhere, they are simple modifiers or general references e.g., "image classification models" - may you do a find and replace to determine what is what here. I wouldn't have expected it to be proper nouns here for example?

- Recipes for **Sparsification from Scratch** usually include the `!GMPruningModifier`, which instructs SparseML to iteratively prune the layers of the model to certain levels (e.g. 80%) over which epochs.

Recipes are then integrated into deep learning training workflows in one of two ways:

#### For Supported Use Cases: CLI

SparseML provides command line scripts that accept recipes as arguments and perform Sparse Transfer Learning and Sparsification from Scratch. We highly
reccomended using the command line scripts. Appending `--help` to the commands demonstrates the full list of arguments.

For example, the following command kicks of **Sparse Transfer Learning** from pre-sparsified YOLOv5 onto the VOC dataset using the pre-made recipes in the SparseZoo:

```bash
$ sparseml.yolov5.train \
    --data VOC.yaml \
    --cfg models_v5.0/yolov5l.yaml \
    --weights zoo:cv/detection/yolov5-l/pytorch/ultralytics/coco/pruned_quant-aggressive_95?recipe_type=transfer \
    --hyp data/hyps/hyp.finetune.yaml \
    --recipe zoo:cv/detection/yolov5-l/pytorch/ultralytics/coco/pruned_quant-aggressive_95?recipe_type=transfer
```

> For example, the following command kicks of **Sparsification of a dense YOLOv5 model from Scratch** using the pre-made recipes in the SparseZoo:

Reviewer: ```suggestion
For example, the following command kicks off **Sparsification of a dense YOLOv5 model from Scratch** using the pre-made recipes in the SparseZoo:
```

```bash
$ sparseml.yolov5.train \  
    --cfg models_v5.0/yolov5l.yaml \
    --weights yolov5l.pt \
    --data coco.yaml \
    --hyp data/hyps/hyp.scratch.yaml \
    --recipe zoo:cv/detection/yolov5-l/pytorch/ultralytics/coco/pruned_quant-aggressive_95
```

For more details on the above as well as examples for more supported use cases see [here](/use-cases/natural-language-processing/question-answering).

#### For Custom Use Cases / Supported Use Cases: Python Integration

The `ScheduledModifierManager` class is used to modify the standard training workflows for both Sparse Transfer Learning
and Sparsification from Scratch. It can be used in PyTorch and TensorFlow/Keras.

The manager classes works by overriding the `model` and `optimizers` to encode sparsity logic.
  Managers can apply recipes in one shot or training aware ways.
  One shot is invoked by calling `.apply(...)` on the manager while training aware requires calls into `initialize(...)` (optional), `modify(...)`, and `finalize(...)`.
  This means only a few lines of code need to be added to begin transfer learning or sparsifying from scratch with pruning and quantization.
  

> For example, the following applies a recipe in a training aware manner:

Reviewer: ```suggestion
For example, the following applies a recipe in a training1aware manner:
```
